From 893a08d546b95679d796625ab3df1fa40843d589 Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Sat, 12 May 2018 07:13:23 -0500
Subject: [PATCH 1262/1936] Signal the spawned Zyn-Fusion process to terminate.

Previously, if using zynaddsubfx in non session manager, Zyn-Fusion would
persist despite stopping the zyn process.
---
 src/main.cpp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 04233b92..9460bdd8 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -28,6 +28,9 @@
 
 #include <getopt.h>
 
+#include <sys/types.h>
+#include <signal.h>
+
 #include <rtosc/rtosc.h>
 #include <rtosc/ports.h>
 #include "Params/PADnoteParameters.h"
@@ -735,7 +738,14 @@ done:
 #endif
 #endif
     }
-
+#ifdef ZEST_GUI
+#ifndef WIN32
+    int ret = kill(gui_pid, SIGHUP);
+    if (ret == -1) {
+        err(1, "Failed to terminate Zyn-Fusion...\n");
+    }
+#endif
+#endif
     exitprogram(config);
     return 0;
 }
-- 
2.47.0

